learningpath-api/src/main/scala/no/ndla/learningpathapi/repository/FolderRepository.scala

```diff
@@ -35,32 +35,35 @@ trait FolderRepository {
     def withTx[T](func: DBSession => T): T =
       DB.localTx { session => func(session) }
 
+    def rollbackOnFailure[T](func: DBSession => Try[T]): Try[T] = {
+      try {
+        DB.localTx { session =>
+          val result = func(session)
+          Success(result.get)
+        }
+      } catch {
+        case throwable: Throwable => Failure(throwable)
+      }
```

Comment:
Jeg kom på at denne vil jo fange alle andre exceptions også. Det gjør sikkert _egentlig_ ikke noe, men vi kan la de passere feks sånn:

```suggestion
      try {
        DB.localTx { session =>
          func(session)
            .recover(ex => throw RollbackException(ex))
        }
      } catch {
        case RollbackException(ex) => Failure(ex)
      }
```